src/dialects/Dialect.ts

```diff
@@ -22,6 +22,20 @@ export interface ITableMetadata {
     comment?: string;
 }
 
+export enum CONSTRAINT_TYPES {
+    c = 'CASCADE',
+    d = 'SET DEFAULT',
+    n = 'SET NULL',
+    r = 'RESTRICT',
+    a = 'NO ACTION',
+}
+
+export interface IColumnMetadataReference {
```

Comment:
We probably don't need to create this interface, this type is already defined in `sequelize`:

```ts
/**
 * References options for the column's attributes
 */
export interface ModelAttributeColumnReferencesOptions {
  /**
   * If this column references another table, provide it here as a Model, or a string
   */
  model?: string | typeof Model;

  /**
   * The column of the foreign table that this column references
   */
  key?: string;

  /**
   * When to check for the foreign key constraing
   *
   * PostgreSQL only
   */
  deferrable?: Deferrable;
}
```